Alerting: Add clean_upgrade config and deprecate force_migration (#78324)
grafana/grafana: docs/sources/setup-grafana/configure-grafana/_index.md

### enabled

> Enable or disable Grafana Alerting. If disabled, all your legacy alerting data will be available again. The default value is `true`.

Reviewer: ```suggestion
Enable or disable Grafana Alerting. If disabled, all legacy alerting data remains available. The default value is `true`.
```

Author: This one I think we should leave as-is, since it would alter the meaning. Before disabling, any existing legacy alerting data is not available, so the act of disabling will make it available again.

Reviewer: @JacobsonMT Ok, sounds good to me, thanks for taking a look and letting me know.

<hr>

## [unified_alerting.upgrade]

> For more information about upgrading to Grafana Alerting, refer to [Upgrade Alerting](/docs/grafana/next/alerting/set-up/migrating-alerts/)

Reviewer: ```suggestion
For more information about upgrading to Grafana Alerting, refer to [Upgrade Alerting](/docs/grafana/next/alerting/set-up/migrating-alerts/.)
```

### clean_upgrade

> When restarting Grafana to upgrade from legacy alerting to Grafana Alerting, first delete any existing Grafana Alerting data from a previous upgrade such as alert rules, contact points, and notification policies. Default is `false`.

Reviewer: ```suggestion
When you restart Grafana to upgrade from legacy alerting to Grafana Alerting, delete any existing Grafana Alerting data from a previous upgrade, such as alert rules, contact points, and notification policies. Default is `false`.
```

> If `false` or unset, existing Grafana Alerting data will not be changed or deleted when switching between legacy and Unified Alerting.

Reviewer: ```suggestion
If `false` or unset, existing Grafana Alerting data is not changed or deleted when you switch between legacy and Unified Alerting.
```

> > **Note.** Should be kept false when not needed as it may cause unintended data-loss if left enabled.

Reviewer: ```suggestion
{{% admonition type="note" %}}
It should be kept false when not needed, as it may cause unintended data loss if left enabled.
{{% /admonition %}}

```
